include/mbgl/shaders/gl/drawable_fill.hpp

```diff
@@ -19,19 +19,13 @@ void main() {
     gl_Position = u_matrix * vec4(a_pos, 0, 1);
```

Comment:
We need to keep the #ifdefs inside the main and the ones that output the varying (in/out). We should remove only the branch that define the uniform, this will go inside the UBO.

Reply:
Example: 
```
#ifndef HAS_UNIFORM_u_color
uniform lowp float u_color_t;
layout (location = 1) in highp vec4 a_color;
out highp vec4 color;
#else
uniform highp vec4 u_color;
#endif
```
should become

```
#ifndef HAS_UNIFORM_u_color
layout (location = 1) in highp vec4 a_color;
out highp vec4 color;
#endif
```
and 
```
uniform lowp float u_color_t;
uniform highp vec4 u_color;
```
will be included in the UBO.

Reply:
The one that is in the main will remain the same:
```
    #ifndef HAS_UNIFORM_u_color
color = unpack_mix_color(a_color, u_color_t);
#else
highp vec4 color = u_color;
#endif
```

Reply:
This is what the code generator does, this fill placeholder was just copied from the background generic program to satisfy the compiler for now.

In the shader manifest JSON, if you include `uses_ubo` it will drop `uniform` decls but keep all the other pre-processor parameters.